### PR TITLE
Fix variable virtual products require a shipping address for Apple Pay and Google Pay

### LINKED ISSUE #6714: Apple Pay: variable virtual products require a shipping address and a choice of a shipping method
### Describe the bug
With Apple Pay (I assume the same may or may not be true for Google Pay), variable virtual products require a shipping address and a choice of shipping method.

### To Reproduce

1. Create a variable product, tick "virtual" checkboxes in all variations.
2. On the product page, click "buy with Apple Pay"
3. A shipping address will be a required field, and a shipping method selection will be presented.
4. If we attempt to purchase a simple virtual product via Apple Pay, the shipping address information/selection is not present and there is no selection of a shipping method.

### Actual behavior
With a virtual variable product, a shipping address selection option will be a displayed in Apple Pay, and a shipping method selection will be presented.

### Screenshots

Variable:
Screenshot 2023-07-08 at 15 40 19

Simple:
Screenshot 2023-07-08 at 15 41 38

### Expected behavior

If a variation is virtual, no shipping address selection should be presented, and no shipping method selection should be offered.

### Desktop (please complete the following information):

* OS: MacOS Ventura 13.4.1 
* Browser Safari 
* Version 16.5.1 (18615.2.9.11.7)

### Additional context
Reported in 6514010-zen

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #6714

#### Changes proposed in this Pull Request
Fixes the issue where Apple Pay and Google Pay require a shipping address for variable virtual products.

Screenshot 2023-09-28 at 13 04 08

#### Testing instructions
- Create a variable product, and tick "virtual" checkboxes in all variations.
- On the product page, select a virtual variation and click "Buy with Apple Pay".
- Verify a shipping address is not required on the modal.

#### Additional context
p1695730786442689-slack-C9976E5MJ

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7317` or branch name `fix/6714-request-button-virtual-variable` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/6714-request-button-virtual-variable) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7f2e39795a5c4325ad954a4c41722291b721bf7f
- Build time: 2023-10-03 11:39:22 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by dpaun1985 — APPROVED** [review]
✅  LGTM. only code review

File: changelog/fix-6714-request-button-virtual-variable [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Virtual variable products require shipping for Apple Pay and Google Pay
```
Comment (anu-rock) [inline changelog/fix-6714-request-button-virtual-variable]:
```suggestion
Virtual variable products no longer require shipping details when checking out with Apple Pay and Google Pay
```

**Review by anu-rock — APPROVED** [review]
LGTM (code-only review since Apple/Google Pay is not available in my region).

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | Not applicable |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | 👍  Almost good (left a comment) |
| Testing instructions  | 👍  Almost good (may be expanded a bit for first-time reviewers) |

**Review by elazzabi — APPROVED** [review]
Tested this locally.

For future reference, you'll need to set up Shipping on WooCommerce before. At least add a zone where your Apple Pay is configured.

Virtual product only - Product page ✅ No shipping address
Virtual product only - checkout page  ✅ No shipping address
Physical product only - checkout page  ✅ Shipping address requested
Virtual & physical products - checkout page  ✅ Shipping address requested

**Pull request merged**